drivers/dma/dma_esp32_gdma.c

```diff
@@ -418,16 +473,27 @@ static int dma_esp32_get_status(const struct device *dev, uint32_t channel,
 		return -EINVAL;
 	}
 
+	memset(status, 0, sizeof(struct dma_status));
+
 	if (dma_channel->dir == DMA_RX) {
 		status->busy = !gdma_ll_rx_is_fsm_idle(data->hal.dev, dma_channel->channel_id);
 		status->dir = PERIPHERAL_TO_MEMORY;
-		status->read_position = dma_channel->desc.dw0.length;
+		desc = (dma_descriptor_t *)gdma_ll_rx_get_current_desc_addr(
+			data->hal.dev, dma_channel->channel_id);
+		if (desc) {
+			status->read_position = desc - dma_channel->desc_list;
+			status->total_copied = desc->dw0.length
+						+ dma_channel->desc_list[0].dw0.size
+						* status->read_position;
+		}
 	} else if (dma_channel->dir == DMA_TX) {
 		status->busy = !gdma_ll_tx_is_fsm_idle(data->hal.dev, dma_channel->channel_id);
 		status->dir = MEMORY_TO_PERIPHERAL;
-		status->write_position = dma_channel->desc.dw0.length;
-		status->total_copied = dma_channel->desc.dw0.length;
-		status->pending_length = dma_channel->desc.dw0.size - dma_channel->desc.dw0.length;
```

Comment:
The reference manual states about the dw0.length field:   
_This field in a transmit descriptor **is written by software** and indicates how many bytes can be read from the buffer_

Therefore I actually think its not possible to calculate the exact pending length. 
I'd also claim that the way it was done previously just returned 0 all the time since `dw0.size` and `dw0.lenth` have both been set to the same value in software